frameworks/Scala/cask/src/main/scala/example/Main.scala

```diff
@@ -0,0 +1,43 @@
+package example
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import com.github.plokhotnyuk.jsoniter_scala.macros._
+import io.undertow.{Undertow, UndertowOptions}
+
+case class Message(message: String)
+
+object Main extends cask.MainRoutes {
+
+  implicit val codec: JsonValueCodec[Message] = JsonCodecMaker.make[Message](CodecMakerConfig())
+
+  override def main(args: Array[String]): Unit = {
+    val server = Undertow.builder
+      .addHttpListener(8080, "0.0.0.0")
+      // increase io thread count as per https://github.com/TechEmpower/FrameworkBenchmarks/pull/4008
+      .setIoThreads(Runtime.getRuntime().availableProcessors() * 2)
```

Comment:
Just curious - did increasing the number of I/O threads improve cask's results for you locally?

It helped undertow's results for me locally, but I still don't know whether it'll do the same in our official benchmarking environment.  We'll have to compare the current run (`06753cdf` on [tfb-status.techempower.com](https://tfb-status.techempower.com/)) to the next run, since the next run will be the first to have the I/O thread changes.

Reply:
Good question - the results for cask are very marginally better when I run with the default number of I/O threads (ie: equal to number of logical cpus): 

```
 Concurrency: 256 for plaintext
 wrk -H 'Host: tfb-server' -H 'Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 256 --timeout 8 -t 12 http://tfb-server:8080/plaintext -s pipeline.lua -- 16
---------------------------------------------------------
Running 15s test @ http://tfb-server:8080/plaintext
  12 threads and 256 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    10.58ms    8.73ms 176.27ms   78.90%
    Req/Sec    25.14k     4.48k  112.97k    93.51%
  Latency Distribution
     50%    8.39ms
     75%   14.44ms
     90%   21.60ms
     99%   39.99ms
  4508576 requests in 15.10s, 554.66MB read
Requests/sec: 298606.82
Transfer/sec:     36.74MB
STARTTIME 1534718350
ENDTIME 1534718365
```

```
 Concurrency: 16 for json
 wrk -H 'Host: tfb-server' -H 'Accept: application/json,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7' -H 'Connection: keep-alive' --latency -d 15 -c 16 --timeout 8 -t 12 "http://tfb-server:8080/json"
---------------------------------------------------------
Running 15s test @ http://tfb-server:8080/json
  12 threads and 16 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   750.23us    2.33ms  39.85ms   93.88%
    Req/Sec     4.79k   737.84    12.61k    71.92%
  Latency Distribution
     50%  156.00us
     75%  262.00us
     90%    0.98ms
     99%   12.98ms
  859100 requests in 15.10s, 122.08MB read
Requests/sec:  56898.63
Transfer/sec:      8.09MB
STARTTIME 1534718216
ENDTIME 1534718231
```

Given there's not much difference it might make sense to submit a change to use the default thread count. Cask uses a `BlockingHandler` which dispatches all work to the task thread pool, which might be why changing the number of I/O threads doesn't seem to do much.

Reply:
Maybe try increasing the worker thread pool size instead of the io thread pool size?